### PR TITLE
Request OpenAlex data

### PR DESCRIPTION
**Pull request by PeterLombaers**
This pull request adds a function to request data from OpenAlex corresponding to a list of identifiers.

### CODE REVIEW

File: asreviewcontrib/datatools/openalex.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    """
+    data = []
+    base_url = "http://api.openalex.org/works"
+    for page_start in range(0, len(id_list), 50):
+        page = id_list[page_start: page_start+50]
+
+        # Set number of results to 50 instead of default 25.
+        url = base_url + f"?filter={id_type}:{'|'.join(page)}&per-page=50"
+        if mailto is not None:
+            url += f"&mailto={mailto}"
```
Comment (J535D165) [inline asreviewcontrib/datatools/openalex.py:42]:
```python 
        params = {
            "filter": '{id_type}:' + '|'.join(id_list[page_start: page_start+50])
            "per-page": 50
        }

        if mailto:
            params["mailto"] = mailto
```
https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls

File: asreviewcontrib/datatools/openalex.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        not necessarily in the same order as the input.
+
+    Raises
+    ------
+    StatusCodeError
+        If a request got a response with status code different from 200.
+    """
+    data = []
+    base_url = "http://api.openalex.org/works"
+    for page_start in range(0, len(id_list), 50):
```
Comment (J535D165) [inline asreviewcontrib/datatools/openalex.py:36]: Please isolate the hardcoded number.

File: asreviewcontrib/datatools/openalex.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        if mailto is not None:
+            url += f"&mailto={mailto}"
+        res = requests.get(url)
+
+        if res.status_code == 200:
+            data += res.json()['results']
+        else:
+            raise StatusCodeError(f"Error in request to Openalex."
+                f"Status code: {res.status_code}. Message: {res.json()}"
+            )
```
Comment (J535D165) [inline asreviewcontrib/datatools/openalex.py:50]:
```
res.raise_for_status()
```

File: asreviewcontrib/datatools/openalex.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import requests
+
+
+class StatusCodeError(Exception):
+    pass
+
+
+def request_by_id(id_list, id_type='doi', mailto=None):
+    """Get the data available in OpenAlex corresonpding to a list of 
+    identifiers.
```
Comment (J535D165) [inline asreviewcontrib/datatools/openalex.py:10]:
```suggestion
    """Get metadata from OpenAlex for list of identifiers.
```

File: asreviewcontrib/datatools/openalex.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    Returns
+    -------
+    list[dict]
+        List of OpenAlex data corresponding to the identifiers, 
+        not necessarily in the same order as the input.
+
+    Raises
+    ------
+    StatusCodeError
+        If a request got a response with status code different from 200.
```
Comment (J535D165) [inline asreviewcontrib/datatools/openalex.py:32]: Requires update

**Review by J535D165 — CHANGES_REQUESTED** [review]
Nice!

Two comments:

- Can we make a CLI for this?
- Built-in support for mixtures of identifiers would be very useful imo

File: asreviewcontrib/datatools/openalex.py (hunk 1 above)
Comment (PeterLombaers) [reply asreviewcontrib/datatools/openalex.py]: I really should've tried reading the readthedocs by now :)

**Comment by J535D165** [comment]
Thanks for the contribution, I added this script to the systematic review datasets repo. I will close this here (for now).

**Pull request closed without merging**